Explorer/Assets/DCL/Landscape/TerrainGenerator.cs

```diff
@@ -0,0 +1,520 @@
+            CreateCliffCornerAt(new Vector3(terrainGenData.terrainSize, 0, 0), Quaternion.Euler(0, 90, 0));
+            CreateCliffCornerAt(new Vector3(0, 0, 0), Quaternion.Euler(0, 180, 0));
+            CreateCliffCornerAt(new Vector3(0, 0, terrainGenData.terrainSize), Quaternion.Euler(0, 270, 0));
+
+            for (var i = 0; i < terrainGenData.terrainSize; i += 16)
+            {
+                Transform side = Object.Instantiate(terrainGenData.cliffSide).transform;
+                side.position = new Vector3(terrainGenData.terrainSize, 0, i + 16);
+                side.rotation = Quaternion.Euler(0, 90, 0);
+                side.SetParent(rootGo.transform, true);
+            }
+
+            for (var i = 0; i < terrainGenData.terrainSize; i += 16)
+            {
+                Transform side = Object.Instantiate(terrainGenData.cliffSide).transform;
+                side.position = new Vector3(i, 0, terrainGenData.terrainSize);
+                side.rotation = Quaternion.identity;
+                side.SetParent(rootGo.transform, true);
+            }
+
+            for (var i = 0; i < terrainGenData.terrainSize; i += 16)
+            {
+                Transform side = Object.Instantiate(terrainGenData.cliffSide).transform;
+                side.position = new Vector3(0, 0, i);
+                side.rotation = Quaternion.Euler(0, 270, 0);
+                side.SetParent(rootGo.transform, true);
+            }
+
+            for (var i = 0; i < terrainGenData.terrainSize; i += 16)
+            {
+                Transform side = Object.Instantiate(terrainGenData.cliffSide).transform;
+                side.position = new Vector3(i + 16, 0, 0);
+                side.rotation = Quaternion.Euler(0, 180, 0);
+                side.SetParent(rootGo.transform, true);
+            }
+        }
+
+        private void CreateCliffCornerAt(Vector3 position, Quaternion rotation)
+        {
+            Transform neCorner = Object.Instantiate(terrainGenData.cliffCorner).transform;
+            neCorner.position = position;
+            neCorner.rotation = rotation;
+            neCorner.SetParent(rootGo.transform, true);
+        }
+
+        private async UniTask SetupEmptyParcelDataAsync()
+        {
+            emptyParcelResult = new NativeHashMap<int2, EmptyParcelData>(emptyParcels.Length, Allocator.Persistent);
+
+            foreach (int2 emptyParcel in emptyParcels)
+                emptyParcelResult.Add(new int2(emptyParcel.x, emptyParcel.y), new EmptyParcelData());
+
+            var job = new SetupEmptyParcels(in emptyParcels, in ownedParcels, ref emptyParcelResult, terrainGenData.heightScaleNerf);
+            JobHandle handle = job.Schedule();
+
+            await handle.ToUniTask(PlayerLoopTiming.Update);
+
+            maxHeightIndex = 0;
+
+            foreach (KVPair<int2, EmptyParcelData> pair in emptyParcelResult)
```

Comment:
Minor: you can do this evaluation in the same `SetupEmptyParcels` job as it's not parallel